src/widgets/help.rs

```diff
@@ -1,261 +1,119 @@
-                    Row::new([
-                        Text::from(vec![Line::from(format!("{} ", m).fg(Color::DarkGray))]),
-                        Text::from(vec![Line::from(format!("{} ", s))]),
-                        Text::from(vec![Line::from(format!("{:?}", a))]),
-                    ])
-                    .fg(config::get().color.base05)
-                    .bg(config::get().color.base00)
-                }
-            })
-            .collect_vec();
-
-        *state.state.selected_mut() = Some(
-            state
-                .state
-                .selected()
-                .unwrap_or_default()
-                .min(rows.len().saturating_sub(1)),
-        );
-
-        let widths = [Constraint::Max(10), Constraint::Max(20), Constraint::Min(0)];
-        let table = Table::new(rows, widths)
-            .header(
-                Row::new(
-                    ["Mode", "Key Chords", "Command"]
-                        .iter()
-                        .map(|h| Text::from(vec![Line::from(h.bold()), "".into()])),
-                )
-                .fg(config::get().color.base05)
-                .bg(config::get().color.base00),
-            )
+        let widths = [Max(10), Max(10), Min(0)];
+        let header = Row::new(["Mode", "Keys", "Command"].map(|h| Line::from(h.bold())))
+            .fg(config::get().color.base05)
+            .bg(config::get().color.base00);
+        let table = Table::new(into_rows(&all_key_bindings), widths)
+            .header(header)
             .column_spacing(5)
             .highlight_symbol(HIGHLIGHT_SYMBOL)
             .highlight_style(config::get().color.base05)
             .highlight_spacing(HighlightSpacing::Always);
         StatefulWidget::render(table, area, buf, &mut state.state);
     }
 }
+
+/// Returns all key bindings for all commands and modes
+///
+/// The result is a vector of tuples containing the mode, command and key bindings joined by a comma
+fn all_key_bindings() -> Vec<(Mode, Command, String)> {
+    ALL_COMMANDS
+        .iter()
+        .flat_map(|(mode, commands)| {
+            commands.iter().map(|command| {
+                let key_bindings = key_bindings_for_command(*mode, *command);
+                let key_bindings = key_bindings.join(", ");
+                (*mode, *command, key_bindings)
+            })
+        })
+        .collect_vec()
+}
+
+/// Returns the key bindings for a specific command and mode
+fn key_bindings_for_command(mode: Mode, command: Command) -> Vec<String> {
+    config::get()
+        .key_bindings
+        .get_config_for_command(mode, command)
+}
+
+/// updates the selected index based on the current mode
+///
+/// Only changes the selected index for the first render
+fn select_by_mode(state: &mut Help, rows: &Vec<(Mode, Command, String)>) {
+    if let Some(mode) = state.mode {
+        tracing::debug!("{:?}", mode);
+        let selected = rows
+            .iter()
+            .find_position(|(mode, _, _)| mode == mode)
```

Comment:
This fixes it for me:

```suggestion
            .find_position(|(m, _, _)| *m == mode)
```